src/Microsoft.AspNet.Mvc.TagHelpers/SelectTagHelper.cs

```diff
@@ -131,7 +121,7 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             // Whether or not (not being highly unlikely) we generate anything, could update contained <option/>
             // elements. Provide selected values for <option/> tag helpers. They'll run next.
-            ViewContext.FormContext.FormData[SelectedValuesFormDataKey] = currentValues;
+            context.Items[typeof(SelectTagHelper)] = currentValues;
```

Comment:
This will need to move to new `Init()` method as part of your reaction PR. Could you rebase that branch on top of this one? Or is that almost ready to get in first?